src/V3Dataflow.cpp

```diff
@@ -0,0 +1,291 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Dataflow based optimization of combinational logic
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2022 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+// V3Dataflow's Transformations:
+//
+//*************************************************************************
+
+#include "config_build.h"
+#include "verilatedos.h"
+
+#include "V3Ast.h"
+#include "V3AstUserAllocator.h"
+#include "V3Dataflow.h"
+#include "V3Error.h"
+#include "V3Global.h"
+#include "V3Graph.h"
+#include "V3UniqueNames.h"
+
+#include "V3Dfg.h"
+#include "V3DfgPasses.h"
+
+#include <vector>
+
+// Extract more combinational logic equations from procedures for better optimization opportunities
+class DataflowExtractVisitor final : public VNVisitor {
```

Comment:
I will restrict this to work on combinational logic (`always_comb`/`always @*` only, as otherwise we might do more work than needed with sparse clock events.

Reply:
Ok

Reply:
Done